renet_webtransport/src/client.rs

```diff
@@ -0,0 +1,186 @@
+
+        read();
+        "
+            .into(),
+        );
+
+        let blob = Blob::new_with_str_sequence_and_options(&script, BlobPropertyBag::new().type_("text/javascript"))?;
+        let url = Url::create_object_url_with_blob(&blob)?;
+        let worker = Worker::new(&url)?;
+        let transfer_array = Array::new_with_length(1);
+        transfer_array.set(0, web_transport.datagrams().readable().into());
+        worker.post_message_with_transfer(&web_transport.datagrams().readable(), &transfer_array)?;
+        let (sender, reciever) = futures_channel::mpsc::unbounded::<Vec<u8>>();
+        let persistent_callback_handle = Self::get_on_msg_callback(sender);
+        worker.set_onmessage(Some(persistent_callback_handle.as_ref().unchecked_ref()));
+
+        let writer = web_transport.datagrams().writable().get_writer()?;
+
+        let (close_sender, close_reciever) = futures_channel::mpsc::unbounded::<bool>();
+        let close_callback_handle: Closure<dyn FnMut(JsValue)> = Self::get_close_callback(close_sender);
+        let _ = web_transport.closed().then(&close_callback_handle).catch(&close_callback_handle);
+
+        Ok(Self {
+            web_transport,
+            persistent_callback_handle,
+            reciever,
+            worker,
+            writer,
+            close_reciever,
+            close_callback_handle,
+            is_disconnected: false,
+        })
+    }
+
+    pub fn update(&mut self, renet_client: &mut RenetClient) {
+        if self.is_disconnected {
+            return;
+        }
+        if let Ok(response) = self.close_reciever.try_next() {
+            self.is_disconnected = response.is_some();
+        }
+        while let Ok(packet) = self.reciever.try_next() {
+            if let Some(packet) = packet {
+                renet_client.process_packet(&packet);
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn send_packets(&mut self, renet_client: &mut RenetClient) {
+        if self.is_disconnected {
+            return;
+        }
+
+        let packets = renet_client.get_packets_to_send();
+        for packet in packets {
+            let data = Uint8Array::new_with_length(packet.len() as u32);
+            data.copy_from(&packet);
+            handle_promise(self.writer.write_with_chunk(&data.into()));
+        }
+    }
+
+    pub fn disconnect(self) {
+        handle_promise(self.writer.close());
+        self.web_transport.close();
+        self.worker.terminate();
+    }
+
+    pub fn is_disconnected(&self) -> bool {
+        self.is_disconnected
+    }
+
+    async fn init_web_transport(url: &str, options: Option<WebTransportOptions>) -> Result<WebTransport, WebTransportError> {
+        let web_transport = match options {
+            Some(options) => WebTransport::new_with_options(url, &options),
+            None => WebTransport::new(url),
+        }?;
+        // returns undefined, once it fullies, webtransport will be ready to use.
```

Comment:
I wanted to justify why we don't expect a value here or why we don't do more stuff with it. So basically when ready resolves it will return undefined and with that we know that the WebTransport is connected.

Since this is the first comment, thank you for the review, really appreciate it that you took the time! :)

Reply:
Yeah that sounds better then what i wanted to commit now("returns undefined when the promise resolves, after that the webtransport will be connected."), thanks! :)